src/pages/Listing/Listing.tsx

```diff
@@ -1,86 +1,76 @@
 import { observer } from "mobx-react-lite";
-import { Link } from "react-router";
+import { Link, useSearchParams } from "react-router";
 
 import { Input } from "@/components/Input";
 import { Button } from "@/components/Button";
 import { RepoCard } from "@/components/RepoCard";
-import { Dropdown, type Option } from "@/components/Dropdown";
+import { Dropdown } from "@/components/Dropdown";
 import { Pagination } from "@/components/Pagination";
 import { IconSearch } from "@/components/Icon/Search";
-import listingPageStore from "@/stores/ListingPageStore";
+import { ListingPageStore, REPO_TYPES } from "@/stores/ListingPageStore";
 import { masks } from "@/config/routesMasks";
 
 import styles from "./Listing.module.scss";
 import { IconLoader } from "@/components/Icon/Loader";
 
-const TYPES: Option[] = [
-  { key: 'all', value: "All" },
-  { key: 'public', value: "Public" },
-  { key: 'private', value: "Private" },
-  { key: 'forks', value: "Forks" },
-  { key: 'sources', value: "Sources" },
-  { key: 'member', value: "Member" },
-];
-
 export const ListingPage: React.FC = observer(() => {
-  const {
-    page, setPage,
-    pages,
-    data,
-    fetch,
-    isLoading,
-  } = listingPageStore;
-
-  const [org, setOrg] = useState("");
-  const [type, setType] = useState<Option>(TYPES[0]);
+  const [searchParams, setSearchParams] = useSearchParams();
 
-  const fetchRepos = () => {
-    listingPageStore.setOrg(org);
-    listingPageStore.setType(type.key);
-
-    fetch();
-  };
+  const storeRef = useRef<ListingPageStore>(null);
+  if (!storeRef.current) {
+    storeRef.current = new ListingPageStore({
+      data: {
+        org: searchParams.get('org'),
+        type: searchParams.get('type'),
+        page: searchParams.get('page')
+      },
+      setSearchParams,
+    });
+  }
+  const store = storeRef.current;
 
   const renderRepos = () => {
-    if (isLoading) return <IconLoader />
+    if (store.isLoading) return <IconLoader />
 
-    return data.map(repo => (
-      <Link className={styles.link} to={masks.repository.create(listingPageStore.org, repo.name)}>
+    return store.data.map(repo => (
+      <Link className={styles.link} to={masks.repository.create(repo.owner, repo.name)}>
         <RepoCard {...repo} />
       </Link>
     ))
   };
 
+  useEffect(() => () => store.destroy());
+
   return <>
     <div className={styles.content}>
       <h1>List of organization repositories</h1>
       <Dropdown
         className={styles.types}
-        options={TYPES}
-        value={type}
+        options={REPO_TYPES}
+        value={store.type}
         placeholder="Type"
-        onChange={setType}
+        onChange={store.setType}
       />
       <div className={styles.search}>
         <Input
           className={styles.input}
-          value={org}
-          onChange={setOrg}
-          onSubmit={() => fetchRepos()}
+          value={store.org}
+          onChange={store.setOrg}
+          onSubmit={() => store.fetch()}

```

Comment:
```suggestion
          onSubmit={store.fetch}
```
Плюс, в случае мемоизирования компонента (React.memo) ссылка будет стабильной и, в отличие от такой записи, не будет создаваться новый колбэк на каждый рендер